pulp_container/app/registry_api.py

```diff
@@ -602,31 +602,31 @@ def mount_blob(self, request, path, repository):
         except models.Blob.DoesNotExist:
             raise BlobNotFound(digest=digest)
 
-        dispatched_task = dispatch(
-            add_and_remove,
-            shared_resources=[version.repository],
-            exclusive_resources=[repository],
-            kwargs={
-                "repository_pk": str(repository.pk),
-                "add_content_units": [str(blob.pk)],
-                "remove_content_units": [],
-            },
+        contains_units = (
+            Q(reserved_resources_record__contains=[get_url(repository)])
+            & Q(reserved_resources_record__contains=[f"shared:{get_url(version.repository)}"])
+            & Q(kwargs__add_content_units__contains=[str(blob.pk)])
         )
+        task = Task.objects.filter(contains_units, name__endswith="add_and_remove").last()
+        if task:
+            if has_task_completed(task):
```

Comment:
is you call `has_task_completed` 1. you will unnecessary wait 1s 2. you will again perform get db query on the task you have just retrieved.